app/src/main/java/org/groundplatform/android/ui/map/gms/features/LineStringOverlayRenderer.kt

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.groundplatform.android.ui.map.gms.features
+
+import android.content.res.Resources
+import com.google.android.gms.maps.GoogleMap
+import javax.inject.Inject
+import org.groundplatform.android.model.geometry.LineString
+import org.groundplatform.android.ui.util.formatDistance
+import org.groundplatform.android.util.midPointToLastSegment
+import org.groundplatform.android.util.tooltipDistanceIfLineStringClosed
+
+class LineStringOverlayRenderer
+@Inject
+constructor(
+  private val resources: Resources,
+  private val tooltipMarkerRenderer: TooltipMarkerRenderer,
+) {
+
+  fun renderOverlayFor(map: GoogleMap, line: LineString) {
+    if (line.coordinates.size < 2) {
+      remove()
+      return
+    }
+    val coords = line.coordinates
+    val distance = coords.tooltipDistanceIfLineStringClosed() ?: return
```

Comment:
It seems you've created this class to define the distance behavior, but I think we agreed that would be up to the ViewModel or business layer to calculate and set; this package only cares about rendering whatever tooltip is set on the `LineString`; it shouldn't know that this particular tooltip is computing using distance. I don't think we need this class at all actually.